test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/awsquerycompatible/AwsQueryCompatibleErrorTest.java

```diff
@@ -62,59 +63,58 @@ public void setupClient() {
     }
 
     @Test
-    public void stubWithHeader_shouldContainResponseMetadata() {
-        stubResponseWithHeaders(queryHeaderValue);
-        AllTypeResponse allTypeResponse = client.allType(SdkBuilder::build);
-        AllTypeResponse allTypeAsyncResponse = asyncClient.allType(SdkBuilder::build).join();
-        verifyResponseMetadata(allTypeResponse, queryHeaderValue);
-        verifyResponseMetadata(allTypeAsyncResponse, queryHeaderValue);
+    public void verifySyncClientException_shouldRetrievedFromHeader() {
+        stubResponseWithQueryHeaderAndBody(QUERY_HEADER_VALUE);
+        try {
+            client.allType(SdkBuilder::build);
+        } catch (AwsServiceException e) {
+            verifyErrorResponse(e, "CustomException");
+        }
     }
 
     @Test
-    public void stubWithNoHeader_responseMetadataShouldBeUnknown() {
-        stubResponseWithoutHeaders();
-        AllTypeResponse allTypesResponse = client.allType(SdkBuilder::build);
-        verifyUnknownResponseMetadata(allTypesResponse);
+    public void verifySyncClientException_shouldRetrievedFromContent() {
+        stubResponseWithQueryHeaderAndBody(INVALID_QUERY_HEADER_VALUE);
+        try {
+            client.allType(SdkBuilder::build);
+        } catch (AwsServiceException e) {
+            verifyErrorResponse(e, "ServiceModeledException");
+        }
     }
 
     @Test
-    public void asyncSubWithHeader_shouldContainResponseMetadata() {
-        stubResponseWithHeaders(queryHeaderValue);
-        AllTypeResponse allTypeAsyncResponse = asyncClient.allType(SdkBuilder::build).join();
-        verifyResponseMetadata(allTypeAsyncResponse, queryHeaderValue);
+    public void verifyAsyncClientException_shouldRetrievedFromHeader() {
+        stubResponseWithQueryHeaderAndBody(QUERY_HEADER_VALUE);
+        try {
+            asyncClient.allType(SdkBuilder::build);
```

Comment:
Missing `fail()` in all test assertions, actually can we use `assertThatThrowBy` instead? 

Example:

https://github.com/aws/aws-sdk-java-v2/blob/5af85b1094334cdabd254fac175b0a789f01d556/services/s3/src/test/java/software/amazon/awssdk/services/s3/functionaltests/CompleteMultipartUploadFunctionalTest.java#L182-L185

Reply:
Cool. Refactored accordingly.